RAN profile work

vitus133 opened the pull request:
@fedepaol @MarSik 
/assign @schseba

vitus133 commented:
@fedepaol @MarSik
/assign @SchSeba

kenyis reviewed (commented):
Are we closer to the CU than this represents?  I am wondering if we leave comments on how to update this for a CU.  Will note the spots.

kenyis commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,22 @@
+# cn-ran-overlays
+
+This folder contains an example configurations for 5G DU site
+(CU - work in progress)
```
I think er should state that a CU is the same profile with the exception that:
1.  Enable a stock kernel instead of an rt-kernel.
2.  Skip PTP
3.  Skip FEC
4.  Add NTP (this is a WIP).

kenyis commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,22 @@
+The [`all-in-one`](all-in-one) directory contains the Kustomize profile for deployment of DU integration features, namely:
+- SCTP MachineConfig patch
+- Performance addon operator and DU performance profile
+- PTP operator and slave profile
```
Optional for a CU

vitus133 replied:
Done folder description

kenyis commented on feature-configs/cn-ran-overlays/all-in-one/cluster/kustomization.yaml:
```diff
@@ -0,0 +1,23 @@
+  - deployment/ptp
+  # PTP profiles
+  - du-common/ptp
+
```
For a CU, we would delete this.  Perhaps we can mark this with a comment.

vitus133 replied:
Done

kenyis commented on feature-configs/cn-ran-overlays/all-in-one/example/performance/performance-profile.yaml:
```diff
@@ -0,0 +1,25 @@
+  numa:
+    topologyPolicy: "restricted"
+  realTimeKernel:
+    enabled: true
```
Comment - set to false for a CU

yuvalk replied:
is that the only change?
Does CU requires isolated CPUs?

kenyis replied:
Yes - a CU requires isolated CPUs.  It will run DPDK workloads.

vitus133 replied:
Added comment

kenyis commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,22 @@
+# cn-ran-overlays
+
+This folder contains an example configurations for 5G DU site
```
Change to RAN sites.

yuvalk commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,22 @@
+- SCTP MachineConfig patch
+- Performance addon operator and DU performance profile
+- PTP operator and slave profile
+- SR-IOV operator and associated profiles
```
what about Local-Storage-Operator ?

kenyis replied:
There is an EPIC for that in 4.8: CNF-1472 Local Persistent Volume Support (without dedicated storage node).  This can be updated under that work.

yuvalk commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,22 @@
+1. Make sure your nodes are labeled as required and MCP is created. If your cluster is brand new, this can be done by cnf-features-deploy/hack/setup-test-cluster.sh
+2. Deploy using the usual project tools:
+
+  `FEATURES_ENVIRONMENT=cn-ran-overlays FEATURES=all-in-one make feature-deploy`
```
that means you wont use the built in features is_ready.sh and post_deploy.sh.
Why cant you explicitly specify the needed features? (and restructure directories to suite that)
it'll also enable you to inherit and reuse deployment info from deploy directory, instead of making a copy of that here that we then need to maintain on every upgrade.

vitus133 replied:
- Right, but I think one of the requirements to the profile is that it can not contain any scripts
- The profile is not feature-centric, but function and site centric. Some of the requirements raised in multiple discussions in our team was to structure the profile in a way that  1. puts everything common for every DU in a single directory ("cluster/du-common" here) 2. separates everything related to the site topology ("site.1.fqdn" here) and 3. separates the hardware-specific details into an "example" directory
- We don't want it to inherit from the deploy directory to improve the stability and not interfere. We are not always using the latest build, trying to reach repeatability in our deployments. The "deploy" directory is updated all the time with different image channels, aiming the latest build. This breaks things from time to time without us knowing where did it come from.
Regarding the need to maintain stuff, I personally was concerned about taking this upstream. I think the profile should live separately until the production methodology (how it is delivered to the customer) is finalized, but in this case I practice "disagree and commit" principle

fedepaol replied:
Agree with @yuvalk . I'd prefer to have the deploy stage centralized and have all the other layers leverage it, so we have control on its evolution. Even if this requires some overhead, it's better than duplicating and having it drift.

Re: the versions of the various operators, each branch points to the related version of the component. Master is now pinned to 4.7, and points to the 4.7 versions. If stability is a concern, you can add an overlay that overrides the version to a more "stable" one until the one related to the current branch is considered stable enough.

fedepaol replied:
Re: usage of scripts.
All the deployment is done via `feature-deploy`, which is a wrapper to kustomize on the various folders.

The other scripts serve the automation, not the user. What the user need to know is only how to use kustomize to deploy the stuff he wants. What we provide (and integrate with the current layout) is a tool to CI to make sure that it can ran tests when the deployment is stable.

fedepaol replied:
Even if the profile is not feature centric, you can leverage the "features" under feature-configs/deploy as much as you do with the new `cluster` customization.

This, with the cost of adding a "is_ready.sh" file for the "all-in-one" feature that returns true when all the needed configurations are applied.

MarSik replied:
And btw, there always will be at least a single script - run kustomize until all is applied.  Because it tries to apply all it can and skips the rest. That means the first execution might install the CRDs (PerformanceProfile), but not the instances (not defined yet) and the second run will create the instances.

yuvalk commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,22 @@
+
+  `FEATURES_ENVIRONMENT=cn-ran-overlays FEATURES=all-in-one make feature-deploy`
+
+3. Wait for the iterations to complete
```
that's what make feature-wait is about
which
a. make it explicit
b. enable notification of failure

@fedepaol maybe it make sense to add a Makefile target that does feature-deploy + feature_wait (without setup and functests).

fedepaol replied:
Don't think so. feature-wait serves CI.
If a user wants to use the features, he just deploys and apply the configuration.

yuvalk replied:
ok
but my point here is also that 
we can't CI the thing without a working feature-wait

fedepaol replied:
Yep, but for CI you also need to do setup and functests  (see my comment below)

yuvalk commented on feature-configs/cn-ran-overlays/all-in-one/cluster/kustomization.yaml:
```diff
@@ -0,0 +1,23 @@
+  # PAO deployment
+  - deployment/performance
+  # PAO profile is given as an example. Must be adapted for a specific hardware
+  - ../example/performance
```
we have an example profile
https://github.com/openshift-kni/cnf-features-deploy/blob/master/feature-configs/typical-baremetal/performance/performance_profile.patch.yaml

kenyis replied:
For customer documentation, to need to keep the RAN profiles in a single space.  This will make the description of how to deploy a CU/DU manually much simpler.  Adding in a separate example adds indirection.

vitus133 replied:
There is a difference between the profiles, and we didn't want to interfere.

fedepaol replied:
+1 on keeping the ran specific configurations explicit and separated.

fedepaol commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,25 @@
+
+## Deployment
+
+1. Make sure your nodes are labeled as required and MCP is created. If your cluster is brand new, this can be done by cnf-features-deploy/hack/setup-test-cluster.sh
```
I'd speak about pre-requisites here.
What we need is a machine config pool and a set of nodes included in that pool.

The `make setup-test-cluster` command will create the machine config pool referenced in the example configuration named "worker-cnf" and will label the nodes accordingly.

MarSik replied:
Btw, plain worker pool is ok as well. There is no need to create additional one. Also there must be at least one unique label on the MCP, because the performance profile can be configured to point to it (instead of trying to be smart).

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
```diff
@@ -0,0 +1,16 @@
+  nicSelector:
+    pfNames: ["ens1f1#0-0"]
+  nodeSelector:
+    node-role.kubernetes.io/site-1-fqdn-worker1: ""
```
This should be mentioned in the readme, same as the label below.
Also, it would make sense to have this labelling automated so we can CI this setup.

Maybe it makes sense to use the same worker-cnf selector?

vitus133 replied:
@josephdrichard

josephdrichard replied:
Yes, we should use the worker-cnf label.  I'll update later today.

vitus133 replied:
@sabinaaledort please see above

vitus133 replied:
@josephdrichard, @fedepaol , @sabinaaledort 
Done in 4911f7

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
```diff
@@ -0,0 +1,16 @@
+  deviceType: vfio-pci
+  isRdma: false
+  nicSelector:
+    pfNames: ["ens1f1#0-0"]
```
This is using one vf only. Is this what is expected? In this way, only one pod will be able to use the vf. @SchSeba wdyt?

fedepaol replied:
@josephdrichard @vitus133 getting back on this, this will expose only one vf.
Is this what we want?

vitus133 replied:
Yes, one VF is fine. No more pods to attach to it

fedepaol replied:
Thanks!

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/site.1.fqdn/networks/sriov-network-mh.yaml:
```diff
@@ -0,0 +1,17 @@
+  name: mh-net-u-site-1-fqdn-worker1
+  namespace: openshift-sriov-network-operator
+spec:
+  ipam:  |
```
I think this can be omitted completely. The driver type is vfio so it won't be needed.

vitus133 replied:
@josephdrichard

josephdrichard replied:
I'll test if it still works with this removed.

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
```diff
@@ -0,0 +1,16 @@
+  name: policy-mh-dpdk-site-1-fqdn-worker1
+  namespace: openshift-sriov-network-operator
+spec:
+  deviceType: vfio-pci
```
This will work with intel only.
Let's document it.
Mellanox must be netdevice + isrdma = true.

vitus133 replied:
@josephdrichard

josephdrichard replied:
This is targeting a specific hardware config, in a lab that it's using intel.   Where would you suggest documenting this?

fedepaol replied:
A note in the docs would work I guess. Something like "sriov configuration notes".
On top of that, a thing you can do is to add a parallel sriov configuration for mellanox, with the same resource name.
At that point, the device plugin will discover only the available ones and will put the capacity on the node accordingly.

It would help us in deploying this on clusters where we have mellanox too.
Finally, I'd document the fact that this configuration depends on the name of the pfs as well, which may differ from cluster to cluster.

vitus133 replied:
@fedepaol I will add this to the README.md

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/site.1.fqdn/kustomization.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Is the addition of the network here worth the complexity of having an extra layer?
This is tied 1:1 with the network policy and with the namespace the workload will run into. 
Would it make sense to have the network too as part of the example layer?

vitus133 replied:
This is because of the assumption that there will be many sites using the same hardware (same NIC port names in the SriovNetworkNodePolicy, example folder ), but different VLANs and / or IP settings (site folders).

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/site.1.fqdn/networks/sriov-network-mh.yaml:
```diff
@@ -0,0 +1,17 @@
+    {
+    }
+  networkNamespace: mh-net-ns-site-1-fqdn-worker1
+  resourceName: mh_u_site_1_fqdn_worker1
```
Can this have the same syntax of the other names and be mh-u-site-1... ?

vitus133 replied:
No, dashes in the resourceName are causing failures

fedepaol replied:
thanks, I was not soure about that.

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
```diff
@@ -0,0 +1,16 @@
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: policy-mh-dpdk-site-1-fqdn-worker1
```
Is there a reasoning behind the naming of the files / resources? If so, can you comment it?

The file here is named policy-netdevice although the deviceType is vfio-pci. Would it make sense to rename the file to be more descriptive (i.e. something referring dpdk)?

vitus133 replied:
@josephrichard

vitus133 replied:
@josephdrichard

josephdrichard replied:
not aware of any reason for this naming.  I agree it should probably be changed.

vitus133 replied:
@josephdrichard, @fedepaol 
Done in 4911f7

fedepaol commented on feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
```diff
@@ -0,0 +1,16 @@
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+  numVfs: 4
+  priority: 10
+  resourceName: mh_u_site_1_fqdn_worker1
```
I'd rename the resource in a way that reflects the characteristics of the device exposed by this policy.
You want to say "I want a vfio device" for this pod rather than something related to it's placement.

vitus133 replied:
@josephdrichard

josephdrichard replied:
Okay, I'll make the change.

fedepaol replied:
Was this left behind? @vitus133 / @josephdrichard

fedepaol commented:
Added a few comments.

One general consideration: since this is going to be a reference, it makes sense to have validation.
Were this profiles tested manually with cnf-tests? I think discovery mode is the best way to do so.

This means, having an automated way to take a new cluster, run a few commands (or even leverage the current `functests-on-ci`) that do the configuration and make sure the setup is ready, and then run the tests.

This can be done applying the configuration and waiting for it to be ready. The `is_ready.sh` pattern is a good way. 
It could even import the `is_ready.sh` files from the `deploy` subfolders so we don't duplicate it.
Last step to remember, is to check that the sriov-configuration is applied correctly. This can be done for example by checking that the nodes have the resource allocatable.

Once this is in place, we need automation. 
@yuvalk / @sabinaaledort would it be possible to set up a baremetal ci like the one we have?

fedepaol commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,25 @@
+# cn-ran-overlays
```
Can the title be more descriptive / start with capital letters?
Also, can you reference this from the main readme so it will be easier to discover ot?

fedepaol commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,25 @@
+1. Make sure your nodes are labeled as required and MCP is created. If your cluster is brand new, this can be done by cnf-features-deploy/hack/setup-test-cluster.sh
+2. Deploy using the usual project tools:
+
+  `FEATURES_ENVIRONMENT=cn-ran-overlays FEATURES=all-in-one make feature-deploy`
```
It makes sense to disconnect from the toolery here. 
Given the current approach of going with a single, non feature centric, profile, I'd specify how to run kustomize to do the deployment, and then add a note saying it can be done also via the makefile rule.

MarSik replied:
+1

yuvalk commented on feature-configs/cn-ran-overlays/all-in-one/cluster/deployment/performance/kustomization.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
```
I think all the deployment stuff can be based on 
https://github.com/openshift-kni/cnf-features-deploy/tree/master/feature-configs/deploy
ie for PAO
add 
```
bases:
  - ../../../../deploy/performance
```

then just kustomize whatever you need (which I think is nothing ATM)

fedepaol replied:
Yep, same thing I proposed in https://github.com/openshift-kni/cnf-features-deploy/pull/377#discussion_r532491759

MarSik replied:
I asked them to decouple this though as there is nothing to gain from the inheritance here. They have a different hardware and a different profile.

MarSik replied:
Oh, scratch this, I see it is the same repo..

yuvalk commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,25 @@
+
+## Overview
+
+The [`all-in-one`](all-in-one) directory contains the Kustomize profile for deployment of DU integration features, namely:
```
we might wanna call it du-profile instead of all-in-one ?

vitus133 replied:
I would rather call it "ran-profile" and then in the README explain that CU is done by this and that modification. Agree?

vitus133 commented:
@fedepaol @yuvalk 
Guys, I think I addressed all the comments (except SR-IOV related that will be done by @josephdrichard)
Could you please have a look?

fedepaol commented on README.md:
```diff
@@ -50,3 +50,7 @@ External tests are consumed as dependencies and ran as part of this same suite.
+
+### Example RAN profile
+
+An example RAN peofile is provided in [feature-configs/cn-ran-overlays](feature-configs/cn-ran-overlays)
```
typo: profile

fedepaol replied:
@vitus133 not sure you saw this, mentioning again. There is a typo "peofile"

fedepaol commented on feature-configs/cn-ran-overlays/ran-profile/cluster/kustomization.yaml:
```diff
@@ -0,0 +1,24 @@
+  # SR-IOV deployment (Networks are deployed as a part of site configuration)
+  - ../../../deploy/sriov
+  # SR-IOV network node policies
+  - ../example/sriov
```
Would it make sense to decoulpe example from cluster?
I feel like example should be a layer on top / on the side of cluster, as opposed to a base of it.
In this way, the deployment will be the same but we'll be able to add another layer for a different set of hw configurations.

fedepaol commented:
@vitus133 looks good!.
Couple of comments: 
I'd like to see a more "layered" approach, with each layer relying on the lower one:
- cluster configuration
- hw configuration
- site specific configuration

In this way, you'll have a better separation of responsibilities of each set of configurations.
Another thing is, I'd pick more descriptive name for each layer. What you explained before is way more clear than example / site name. I'd go for "hw configuration" / "site specific configuration" or something like that, and I'd mention the structure in the readme, clarifying that this is for example purpouses, referring to a configuration wehre all the sites have the same hw structure but they may differ in terms of network topology.

fedepaol commented:
Another thing is (but this can be addressed in a separate PR), to make this CI-able we need to drop a `is_ready.sh` script that returns true when all the configuration is in place and we can launch the tests.

vitus133 commented:
> I'd like to see a more "layered" approach, with each layer relying on the lower one:
> 
>     * cluster configuration
> 
>     * hw configuration
> 
>     * site specific configuration
> 
> 
> In this way, you'll have a better separation of responsibilities of each set of configurations.
> Another thing is, I'd pick more descriptive name for each layer. What you explained before is way more clear than example / site name. I'd go for "hw configuration" / "site specific configuration" or something like that, and I'd mention the structure in the readme, clarifying that this is for example purpouses, referring to a configuration wehre all the sites have the same hw structure but they may differ in terms of network topology.

@fedepaol I agree that the manifest hierarchy seems to be suboptimal, but there is a work still ongoing to refine it. Could it be possible to postpone this issue to the next PR?

josephrichard commented:
Can someone advise why I'm on this thread Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Joseph Richard <notifications@github.com> Date: 12/1/20  12:42 PM  (GMT-05:00) To: openshift-kni/cnf-features-deploy <cnf-features-deploy@noreply.github.com> Cc: josephrichard <joseph.richard90@gmail.com>, Mention <mention@noreply.github.com> Subject: Re: [openshift-kni/cnf-features-deploy] RAN profile work (#377) 
@josephdrichard commented on this pull request.



In feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
> @@ -0,0 +1,16 @@
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: policy-mh-dpdk-site-1-fqdn-worker1

not aware of any reason for this naming.  I agree it should probably be changed.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

josephdrichard commented:
> Can someone advise why I'm on this thread Sent from my Verizon, Samsung Galaxy smartphone -------- Original message --------From: Joseph Richard <notifications@github.com> Date: 12/1/20 12:42 PM (GMT-05:00) To: openshift-kni/cnf-features-deploy <cnf-features-deploy@noreply.github.com> Cc: josephrichard <joseph.richard90@gmail.com>, Mention <mention@noreply.github.com> Subject: Re: [openshift-kni/cnf-features-deploy] RAN profile work (#377) @josephdrichard commented on this pull request. In feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
> @@ -0,0 +1,16 @@
> +apiVersion: sriovnetwork.openshift.io/v1 +kind: SriovNetworkNodePolicy +metadata: + name: policy-mh-dpdk-site-1-fqdn-worker1 not aware of any reason for this naming. I agree it should probably be changed. —You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

You must have been tagged by mistake instead of me somewhere.

vitus133 commented:
Sorry, this is by mistake. Will try to remove...

On Wed, Dec 2, 2020 at 1:18 AM josephrichard <notifications@github.com>
wrote:

> Can someone advise why I'm on this thread Sent from my Verizon, Samsung
> Galaxy smartphone
> -------- Original message --------From: Joseph Richard <
> notifications@github.com> Date: 12/1/20 12:42 PM (GMT-05:00) To:
> openshift-kni/cnf-features-deploy <cnf-features-deploy@noreply.github.com>
> Cc: josephrichard <joseph.richard90@gmail.com>, Mention <
> mention@noreply.github.com> Subject: Re:
> [openshift-kni/cnf-features-deploy] RAN profile work (#377)
> @josephdrichard commented on this pull request.
>
>
>
> In
> feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml:
> > @@ -0,0 +1,16 @@
> +apiVersion: sriovnetwork.openshift.io/v1
> +kind: SriovNetworkNodePolicy
> +metadata:
> + name: policy-mh-dpdk-site-1-fqdn-worker1
>
> not aware of any reason for this naming. I agree it should probably be
> changed.
>
> —You are receiving this because you were mentioned.Reply to this email
> directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#issuecomment-736882711>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYDXBWBD2TNXKK5O7FMBWLSSV2UFANCNFSM4UFPWWEA>
> .
>

fedepaol commented:
> > I'd like to see a more "layered" approach, with each layer relying on the lower one:
> > ```
> > * cluster configuration
> > 
> > * hw configuration
> > 
> > * site specific configuration
> > ```
> > 
> > 
> > In this way, you'll have a better separation of responsibilities of each set of configurations.
> > Another thing is, I'd pick more descriptive name for each layer. What you explained before is way more clear than example / site name. I'd go for "hw configuration" / "site specific configuration" or something like that, and I'd mention the structure in the readme, clarifying that this is for example purpouses, referring to a configuration wehre all the sites have the same hw structure but they may differ in terms of network topology.
> 
> @fedepaol I agree that the manifest hierarchy seems to be suboptimal, but there is a work still ongoing to refine it. Could it be possible to postpone this issue to the next PR?

@vitus133 the thing that bothers me the most is the dependency between cluster and example.
This is because they have different responsibilities, and ideally example depends on cluster (and not the other way around).

A quick solution is to have the `example` folders as a direct bases of `ran-profile` (instead of having them part of `cluster`). 
Do you think this is doable?

In any case, can you rename the folder and document as above? I think it will make the whole layout more understandable.

fedepaol commented:
@vitus133 another ask: before we push this, can you squash the commits and come up with a descriptive commit message?

yuvalk commented:
the commits situation must also be resolved. 
we cant merge this with 11 arbitrary commits.

@vitus133 - I can sync help you on amending this. should be fairly easy

yuvalk commented on feature-configs/cn-ran-overlays/ran-profile/example/performance/kustomization.yaml:
```diff
@@ -0,0 +1,6 @@
+kind: Kustomization
+
+resources:
+  - performance-profile.yaml
```
shouldnt that be a strategicMerge?

vitus133 commented:
Completely restructured and squashed

vitus133 commented:
Yep, one VF is fine. No more pods to attach to

On Wed, Dec 2, 2020 at 3:04 PM Federico Paolinelli <notifications@github.com>
wrote:

> *@fedepaol* commented on this pull request.
> ------------------------------
>
> In
> feature-configs/cn-ran-overlays/all-in-one/example/sriov/sriov-network-node-policy-netdevice-mh.yaml
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#discussion_r534150227>
> :
>
> > @@ -0,0 +1,16 @@
> +apiVersion: sriovnetwork.openshift.io/v1
> +kind: SriovNetworkNodePolicy
> +metadata:
> +  name: policy-mh-dpdk-site-1-fqdn-worker1
> +  namespace: openshift-sriov-network-operator
> +spec:
> +  deviceType: vfio-pci
> +  isRdma: false
> +  nicSelector:
> +    pfNames: ["ens1f1#0-0"]
>
> @josephdrichard <https://github.com/josephdrichard> @vitus133
> <https://github.com/vitus133> getting back on this, this will expose only
> one vf.
> Is this what we want?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#discussion_r534150227>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYDXBTKWRPQJBS7BYE4F5LSSY3MJANCNFSM4UFPWWEA>
> .
>

fedepaol commented on feature-configs/cn-ran-overlays/ran-profile/site.1.fqdn/networks/sriov-network-node-policy-dpdk-mh.yaml:
```diff
@@ -0,0 +1,21 @@
+  isRdma: false
+  nicSelector:
+# The exact physical function name must match the hardware used    
+    pfNames: ["ens1f1#0-0"]
```
There's no point in partitioning the pf here, as it will be only used for dpdk

fedepaol commented:
Only one nit about pf partitioning, then looks good on my side.
@yuvalk please take a look.

Before approving, please confirm that this was validated locally

yuvalk commented on feature-configs/cn-ran-overlays/ran-profile/cluster-config/du-common/performance/performance-profile.yaml:
```diff
@@ -0,0 +1,26 @@
+  additionalKernelArgs:
+  - nosmt
+  cpu:
+    # Workaround for RT kernel bugs that consume too much CPU power
```
not sure what part here is the workaround?

fedepaol commented:
@vitus133 can you squash the commits?

vitus133 commented:
@fedepaol  Have I screwed up? I see Sebastian's commit in my PR...

fedepaol commented:
> @fedepaol Have I screwed up? I see Sebastian's commit in my PR...

yep not sure why but there's a commit that was merged. Did you rebase this pr?

vitus133 commented:
I squashed, one commit too many

On Wed, Dec 2, 2020 at 5:59 PM Federico Paolinelli <notifications@github.com>
wrote:

> @fedepaol <https://github.com/fedepaol> Have I screwed up? I see
> Sebastian's commit in my PR...
>
> yep not sure why but there's a commit that was merged. Did you rebase this
> pr?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#issuecomment-737323232>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYDXBR4E6DRSTANZPV6X3LSSZP6DANCNFSM4UFPWWEA>
> .
>

vitus133 commented:
I think it's irreversible...
Can it go in like that? Or I can make a clean branch and create another PR.
Please let me know.
Thank you,
VG

On Wed, Dec 2, 2020 at 5:59 PM Federico Paolinelli <notifications@github.com>
wrote:

> @fedepaol <https://github.com/fedepaol> Have I screwed up? I see
> Sebastian's commit in my PR...
>
> yep not sure why but there's a commit that was merged. Did you rebase this
> pr?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#issuecomment-737323232>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYDXBR4E6DRSTANZPV6X3LSSZP6DANCNFSM4UFPWWEA>
> .
>

josephdrichard commented:
> I think it's irreversible... Can it go in like that? Or I can make a clean branch and create another PR. Please let me know. Thank you, VG
> […](#)
> On Wed, Dec 2, 2020 at 5:59 PM Federico Paolinelli ***@***.***> wrote: @fedepaol <https://github.com/fedepaol> Have I screwed up? I see Sebastian's commit in my PR... yep not sure why but there's a commit that was merged. Did you rebase this pr? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#377 (comment)](https://github.com/openshift-kni/cnf-features-deploy/pull/377#issuecomment-737323232)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABYDXBR4E6DRSTANZPV6X3LSSZP6DANCNFSM4UFPWWEA> .

You should be able to force push to the lack:ran-profile branch to update the pull request.

fedepaol commented:
Looks good on my side. Thanks for reworking this!

@yuvalk thoughts?
@vitus133 before merging, can you confirm this was validated (as we don't have CI blocking this pr)?

yuvalk commented:
/test e2e-gcp-ovn

yuvalk commented:
> Looks good on my side. Thanks for reworking this!
> 
> @yuvalk thoughts?
Yes
I have some comments on the general structure of the kustomization folders.

I think it makes more sense in feature-configs/cn-ran-overlays/ran-profile/cluster-config/kustomization.yaml
to use something like 
```
bases:
  ...
  - du-common/sctp
```

and then in that folder to have
```
bases:
  - ../../../../../deploy/sctp
```

this will allow future customization by customers with their own overlay, instead of working through the internal structure to realize, for example, they need to edit deploy/sctp to change the mcp label.

In general, I think we should strive for this general approach - that the profile is fixed, and changes are made with external overlays.
I think the general approach to that should be that the site directory is based on the cluster, then you can apply -k that and get the whole thing.
then user can create his own directory overriding the default site.

also some of the naming might require additional comment or better phrasing like mh-net-u-site-1-fqdn-worker1 - not sure how many people will understand u = user plance.

but all these comments can be addressed in future PRs,
so I'll give my lgtm
and once we have a confirmation that this is validated, can be approved IMHO.

> @vitus133 before merging, can you confirm this was validated (as we don't have CI blocking this pr)?

yuvalk commented:
/lgtm

kenyis commented:
@SchSeba 

We have 2 approvals for this PR: @yuvalk and @fedepaol .  This is profile that we are currently using for our PTP work.  We will continue to test the performance of this profile under CNF-950.  We will test SRIOV under CNF-1398 and SCTP under CNF-1405.  This will get plenty of soak.

Can you approve this please?

/KenY

fedepaol commented:
> @SchSeba
> 
> We have 2 approvals for this PR: @yuvalk and @fedepaol . This is profile that we are currently using for our PTP work. We will continue to test the performance of this profile under CNF-950. We will test SRIOV under CNF-1398 and SCTP under CNF-1405. This will get plenty of soak.
> 
> Can you approve this please?
> 
> /KenY

Before approving we just confirmation that the setup as a whole is validated as @yuvalk mentioned here https://github.com/openshift-kni/cnf-features-deploy/pull/377#issuecomment-737783058 , as this is not covered by CI.

vitus133 commented:
The branch was tested on 4.6 cluster with the official 4.6 operator images. The difference since than is only in reliance on the upstream 4.7 operator images. This must be tested on 4.7 cluster, which I currently don't have access to.

The link to the demo is here:
https://drive.google.com/file/d/1t1mtmDpwBar8RIk4WapDfb0c-dVcLtTf/view?usp=sharing

yuvalk commented:
> The branch was tested on 4.6 cluster with the official 4.6 operator images. The difference since than is only in reliance on the upstream 4.7 operator images. This must be tested on 4.7 cluster, which I currently don't have access to.
> 
> The link to the demo is here:
> https://drive.google.com/file/d/1t1mtmDpwBar8RIk4WapDfb0c-dVcLtTf/view?usp=sharing

@vitus133 @kenyis 
this is the u/s discussion
I think it's better to refrain from referencing internal content here (CNF Jira tickets, internal drive content etc.)
if you can please remove these (then I'll remove this comment)

@vitus133 make sure to keep the content though on the fact that this PR is fully validated, which is important 

TIA

vitus133 commented:
Done, sorry...

On Thu, Dec 3, 2020 at 5:17 PM Yuval Kashtan <notifications@github.com>
wrote:

> The branch was tested on 4.6 cluster with the official 4.6 operator
> images. The difference since than is only in reliance on the upstream 4.7
> operator images. This must be tested on 4.7 cluster, which I currently
> don't have access to.
>
> The link to the demo is here:
>
> https://drive.google.com/file/d/1t1mtmDpwBar8RIk4WapDfb0c-dVcLtTf/view?usp=sharing
>
> @vitus133 <https://github.com/vitus133> @kenyis
> <https://github.com/kenyis>
> this is the u/s discussion
> I think it's better to refrain from referencing internal content here (CNF
> Jira tickets, internal drive content etc.)
> if you can please remove these (then I'll remove this comment)
>
> @vitus133 <https://github.com/vitus133> make sure to keep the content
> though on the fact that this PR is fully validated, which is important
>
> TIA
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#issuecomment-738071025>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYDXBTHENPHBGS6MSALSQLSS6TWZANCNFSM4UFPWWEA>
> .
>

kenyis commented:
Note that I cleaned up my comment as well.  Good to know.

fedepaol commented:
Thanks for this!
/approve

StephenJamesSmith commented on feature-configs/cn-ran-overlays/README.md:
```diff
@@ -0,0 +1,113 @@
+
+## Overview
+
+Current example is focused on a DU. A CU profile can be built by
```
Shouldn't this say "A DU profile..."?

vitus133 commented:
No, what I meant here is that you can build a CU profile by taking the DU
profile as an example and doing the changes described below.

On Fri, Dec 11, 2020 at 2:59 PM Stephen James Smith <
notifications@github.com> wrote:

> *@StephenJamesSmith* commented on this pull request.
> ------------------------------
>
> In feature-configs/cn-ran-overlays/README.md
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#discussion_r540929433>
> :
>
> > +Running cell application on COTS hardware requires following features to be enabled:
> +
> +- Real-time kernel
> +- CPU isolation
> +- NUMA awareness
> +- HUGEPAGES memory management
> +- Precision timing synchronization using PTP
> +- AVX-512 instruction set (for Flexran and / or FPGA implementation)
> +- Additional features depending on the RAN operator requirements
> +
> +Accessing hardware acceleration devices and high throughput network interface cards by virtualized software applications requires use of SRIOV and Passthrough PCI device virtualization.
> +In addition to the compute and acceleration requirements, DUs operate on multiple internal and external networks.
> +
> +## Overview
> +
> +Current example is focused on a DU. A CU profile can be built by
>
> Shouldn't this say "A DU profile..."?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift-kni/cnf-features-deploy/pull/377#pullrequestreview-550075901>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYDXBRQGCTSXU53AV6M3I3SUIJSZANCNFSM4UFPWWEA>
> .
>
